qiskit/primitives/backend_estimator_dev.py

```diff
@@ -0,0 +1,652 @@
+        job.submit()
+        return job
+
+    def _compute(
+        self,
+        circuits: tuple[QuantumCircuit, ...],
+        observables: tuple[BaseOperator | PauliSumOp, ...],
+        parameter_values: tuple[tuple[float, ...], ...],
+        **run_options,
+    ) -> EstimatorResult:
+        """Solve expectation value problem on backend."""
+        # Pre-processing
+        circuit_bundles = self._preprocess(circuits, observables, parameter_values)
+        metadata_bundles = tuple(tuple(c.metadata for c in bun) for bun in circuit_bundles)
+        job_circuits = list(c for bun in circuit_bundles for c in bun)  # TODO: accept tuple
+
+        # Raw results: counts
+        counts_list: list[Counts] = self._run_on_backend(job_circuits, **run_options)
+
+        # Post-processing
+        counts_iter = iter(counts_list)
+        counts_bundles = tuple(tuple(next(counts_iter) for _ in bun) for bun in circuit_bundles)
+        expval_list, var_list, shots_list = self._postprocess(counts_bundles, metadata_bundles)
+
+        # Results
+        values = np.real_if_close(expval_list)
+        metadata = [
+            {"variance": var, "shots": shots, "num_circuits": len(circuit_bundle)}
+            for var, shots, circuit_bundle in zip(var_list, shots_list, circuit_bundles)
+        ]
+        return EstimatorResult(values, metadata)
+
+    def _preprocess(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        observables: Sequence[BaseOperator | PauliSumOp],
+        parameter_values: Sequence[Sequence[float]],
+    ) -> tuple[tuple[QuantumCircuit, ...], ...]:
+        """Preprocess circuit-observable experiments to runnable tuples of circuits, one per pair."""
+        return tuple(
+            self._preprocess_single(circuit, observable, params)
+            for circuit, observable, params in zip(circuits, observables, parameter_values)
+        )
+
+    def _preprocess_single(
+        self,
+        circuit: QuantumCircuit,
+        observable: BaseOperator,
+        parameter_values: Sequence[float],
+    ) -> tuple[QuantumCircuit, ...]:
+        """Preprocess single circuit-observable experiment to runnable tuple of circuits."""
+        circuit = self._transpile(circuit)  # TODO: Cache (produces a copy)
+        circuit.assign_parameters(parameter_values, inplace=True)
+        circuit = self._run_bound_pass_manager(circuit)
+        measurements = self._build_measurement_circuits(observable)
+        circs_w_meas = self._compose_measurements(circuit, measurements)
+        return circs_w_meas
+
+    def _run_on_backend(self, circuits: list[QuantumCircuit], **run_options) -> list[Counts]:
+        """Run circuits on backend bypassing max circuits allowed."""
+        # Max circuits
+        total_circuits: int = len(circuits)
+        max_circuits: int = getattr(self.backend, "max_circuits", None) or total_circuits
+
+        # Raw results
+        jobs: tuple[Job] = tuple(
```

Comment:
No type conversion is happening in this line

Reply:
Yes. I know. But it is simply difficult to read. It's not Pythonic.

Reply:
Original:
```python
jobs = [
    backend.run(circuits[pos : pos + max_circuits], **run_options)
    for pos in range(0, len(circuits), max_circuits)
]
```

Refactor:
```python
jobs: tuple[Job] = tuple(
    self.backend.run(circuits[split : split + max_circuits], **run_options)
    for split in range(0, total_circuits, max_circuits)
)
```

I don't see much of an stylistic difference other than the type annotation, and use of an immutable data structure instead of a mutable one (since the resulting instance should not be altered). Not sure how this makes it less pythonic, but what would you propose instead?

Reply:
The default in the language specification is list, and there is no reason to include unnecessary conversions. I also find the latter more difficult because I have to convert it once in my head.